bin/node-template/node/src/service.rs

```diff
@@ -206,69 +195,53 @@ pub fn new_full(config: Configuration) -> Result<TaskManager, ServiceError> {
 
 /// Builds a new service for a light client.
 pub fn new_light(config: Configuration) -> Result<TaskManager, ServiceError> {
-	let inherent_data_providers = InherentDataProviders::new();
-
-	ServiceBuilder::new_light::<Block, RuntimeApi, Executor>(config)?
-		.with_select_chain(|_config, backend| {
-			Ok(LongestChain::new(backend.clone()))
-		})?
-		.with_transaction_pool(|builder| {
-			let fetcher = builder.fetcher()
-				.ok_or_else(|| "Trying to start light transaction pool without active fetcher")?;
+	let (client, backend, keystore, mut task_manager, on_demand) =
+		sc_service::new_light_parts::<Block, RuntimeApi, Executor>(&config)?;
+	
+	let transaction_pool_api = Arc::new(sc_transaction_pool::LightChainApi::new(
+		client.clone(), on_demand.clone(),
+	));
+	let transaction_pool = sc_transaction_pool::BasicPool::new_light(
+		config.transaction_pool.clone(),
+		transaction_pool_api,
+		config.prometheus_registry(),
+		task_manager.spawn_handle(),
+	);
+
+	let grandpa_block_import = sc_finality_grandpa::light_block_import(
+		client.clone(), backend.clone(),
+		Arc::new(on_demand.checker().clone()) as Arc<_>,
+	)?;
+	let finality_proof_import = grandpa_block_import.clone();
+	let finality_proof_request_builder =
+		finality_proof_import.create_finality_proof_request_builder();
+
+	let import_queue = sc_consensus_aura::import_queue::<_, _, _, AuraPair, _>(
+		sc_consensus_aura::slot_duration(&*client)?,
+		grandpa_block_import,
+		None,
+		Some(Box::new(finality_proof_import)),
+		client.clone(),
+		InherentDataProviders::new(),
+		&task_manager.spawn_handle(),
+		config.prometheus_registry(),
+	)?;
+
+	let finality_proof_provider = Arc::new(GrandpaFinalityProofProvider::new(
+		backend.clone(), client.clone() as Arc<_>
+	));
+
+	sc_service::build(sc_service::ServiceParams {	
```

Comment:
Can we not really get rid off `ServiceParams`?

Reply:
And replace it with what? Given that there are a lot of parameters to pass to the builder, I'd much rather have them all in one struct so that they can be named.

Reply:
I think the issue is that the number of arguments vary with the implementations. So we can't really create a "one good struct do all" for this. It needs to be more flexible than there. Thus the tuple instead of a struct.

Reply:
Related conversation: https://github.com/paritytech/substrate/pull/6557#discussion_r449416040

Reply:
What we could do is possibly have `FullServiceParams` and `LightServiceParams` that you can call `into()` on and return the full service params. That might make it easier to cut out some of those fields.

Reply:
The service should just expose the methods that will be called by the builder internally. Than you can call whatever methods you like in the concrete service implementation.

Reply:
Let's save this for another PR.